### PR TITLE
Get rid of QtGui dependency in networking library.

### PR DESCRIPTION
**Pull request by namark**
Builds upon https://github.com/vircadia/vircadia/pull/1620

The dependency on QtGui mostly came from shared library, so in this PR all the GUI dependent code has been moved out, mostly verbatim to a new library called shared-gui, but in a few cases partially or to the point of use with some modifications:

1. `shared/src/GLMHelpers` has been split into two parts, the bulk of it remains in `shared` library, but a small part (two conversion functions that involve `QColor` and `QMatrix4x4`) moved to `shared-gui/src/shared/gui` under the same name.
2. `networking/src/AddressManager` clipboard functionality has been moved to `interface/src/Menu`, which was the only use case (Navigate -> Copy Address/Path to Clipboard).
3. `shared/src/shared/AbstractLoggerInterface::locateLog` function has been replaced with `getLocation` function that just returns a string, with the locating functionality (`FileUtils::locateFile`) moved to the only use case in `interface/src/ui/LogDialog` (Developer -> Log -> Reveal log file).

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-74e0cda-.tar_1fb4be9d.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-74e0cda-_1fb4be9d.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-74e0cda-.tar_1fb4be9d.gz

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-060b6e6-.tar_0aabc668.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-060b6e6-.tar_0aabc668.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-060b6e6-_0aabc668.sh

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-Interface-PR1622-7249256-_c149ed1d.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-7249256-.tar_c149ed1d.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-7249256-.tar_c149ed1d.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-7249256-_c149ed1d.sh

**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-7249256-_c149ed1d.dmg

**build (windows-2019, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-7249256-_c149ed1d.exe

File: libraries/shared-gui/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+set(TARGET_NAME shared-gui)
+
+include_directories("${QT_DIR}/include/QtCore/${QT_VERSION}/QtCore" "${QT_DIR}/include/QtCore/${QT_VERSION}")
+
+# TODO: there isn't really a good reason to have Script linked here - let's get what is requiring it out (RegisteredMetaTypes.cpp)
```
Comment (ctrlaltdavid) [inline libraries/shared-gui/CMakeLists.txt:5]: Is this TODO still valid?

File: libraries/shared-gui/src/shared/gui/GLMHelpers.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+//
+//  GLMHelpers.cpp
+//  libraries/shared-gui/src/shared/gui
+//
+//  Created by Stephen Birarda on 2014-08-07.
+//  Copyright 2014 High Fidelity, Inc.
```
Comment (ctrlaltdavid) [inline libraries/shared-gui/src/shared/gui/GLMHelpers.cpp:6]: Is this correct?

File: libraries/shared-gui/src/shared/gui/GLMHelpers.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+#include "GLMHelpers.h"
+
+
+glm::vec4 toGlm(const QColor& color) {
+    return glm::vec4(color.redF(), color.greenF(), color.blueF(), color.alphaF());
+}
+
+QMatrix4x4 fromGlm(const glm::mat4 & m) {
+  return QMatrix4x4(&m[0][0]).transposed();
```
Comment (ctrlaltdavid) [inline libraries/shared-gui/src/shared/gui/GLMHelpers.cpp:20]:
```suggestion
    return QMatrix4x4(&m[0][0]).transposed();
```

File: libraries/shared-gui/src/shared/gui/GLMHelpers.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
+//
+//  GLMHelpers.h
+//  libraries/shared-gui/src/shared/gui
+//
+//  Created by Stephen Birarda on 2014-08-07.
+//  Copyright 2014 High Fidelity, Inc.
```
Comment (ctrlaltdavid) [inline libraries/shared-gui/src/shared/gui/GLMHelpers.h:6]: Is this correct?

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]
The moved files, e.g., AABox.cpp need the paths in their headers updated (e.g., "libraries/octree/src" ==> "libraries/shared-gui/src".

File: libraries/shared-gui/CMakeLists.txt (hunk 1 above)
Comment (namark) [reply libraries/shared-gui/CMakeLists.txt:5]: I believe so, both `shared-gui` and `shared` still depend on QScript.

File: libraries/shared-gui/src/shared/gui/GLMHelpers.cpp (hunk 2 above)
Comment (namark) [reply libraries/shared-gui/src/shared/gui/GLMHelpers.cpp:6]: I just copied part of the code, so I think it makes sense to leave the copyright statement untouched. Technically I created the file, but it was just a part of moving things around, and not creating something new.

**Comment by namark** [comment]
Paths in header comments in shared-gui library fixed.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-f6563cf-_80f36a5b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-f6563cf-.tar_80f36a5b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-f6563cf-.tar_80f36a5b.Z

**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-Interface-PR1622-f6563cf-_80f36a5b.dmg

**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-f6563cf-_80f36a5b.dmg

**build (windows-2019, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1622/Vircadia-PR1622-f6563cf-_80f36a5b.exe

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by ctrlaltdavid** [comment]
Tested on Windows:
- Navigate > Copy Address to Clipboard
- Navigate > Copy Path to Clipboard
- Developer > Log > Reveal log file

Both domain server and Interface appear to run OK.

**Comment by daleglass** [comment]
Everything looks good on Linux

**Pull request merged**